nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,65 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+router.get("/", async (request, response) => {
+  try {
+    console.log(meals);
+    console.log("in /api/meals");
+    let selectedMeals = meals;
+    if ('maxPrice' in request.query) {
+      const maxPrice = parseInt(request.query.maxPrice);
+      if (isNaN(maxPrice)) {
+        res.status(400).send({ error: "maxPrice should be an integer" });
+        return;
+      }
+      selectedMeals = selectedMeals.filter(meal => meal.price < maxPrice);
+    }
+    if ('title' in request.query) {
+      const title = request.query.title;
+      selectedMeals = selectedMeals.filter(meal => meal.title.toLowerCase().includes(title.toLowerCase()));
+    }
+    if ('createdAfter' in request.query) {
+      const createdAfter = new Date(request.query.createdAfter);
+      if (!createdAfter.getDate()) {
+        response.status(404).send('must be a valid date.');
+        return;
+      }
+      selectedMeals = selectedMeals.filter(meal => new Date(meal.createdAt).getDate() > createdAfter.getDate());
+    }
+    if ('limit' in request.query) {
+      const limit = parseInt(request.query.limit);
+      if (isNaN(limit)) {
+        res.status(400).send({ error: "limit should be an integer" });
+        return;
+      }
+      selectedMeals = selectedMeals.slice(0, limit);
+    }
+  
+    response.send(selectedMeals);
+  } catch (error) {
+    throw error;
```

Comment:
throwing an error here is not helpful because it would crash your server app if it's not caught.
you can return an error code like` 500 Internal Server Error instead`